packages/embedded-wallet/src/ui/slider/Slider.test.tsx

```diff
@@ -0,0 +1,129 @@
+  );
+
+  return refreshSliderFrom({sliderWrapper} as MockSlider);
+};
+
+const refreshSliderFrom = (slider: MockSlider) => {
+  const {sliderWrapper} = slider;
+
+  sliderWrapper.update();
+
+  return {
+    sliderWrapper: slider.sliderWrapper,
+    minLabelElement: sliderWrapper.find(`label.${css.min}`),
+    sliderElement: sliderWrapper.find("input[type='range']"),
+    maxLabelElement: sliderWrapper.find(`label.${css.max}`),
+    valueLabelElement: sliderWrapper.find(`label.${css.value}`)
+  };
+};
+
+const triggerKeyOn = (component: MockSlider, element: keyof MockSlider, key: string) => {
+  component[element].simulate('keydown', {key});
+  return refreshSliderFrom(component);
+};
+
+describe('UI - Slider', () => {
+  let component: MockSlider;
+
+  beforeEach(() => {
+    component = mockSlider();
+  });
+
+  it('can be instantiated', () => {
+    expect(component.minLabelElement.exists()).toEqual(true);
+    expect(component.minLabelElement.text()).toEqual('0');
+    expect(component.minLabelElement.prop('aria-hidden')).toEqual('true');
+    expect(component.valueLabelElement.exists()).toEqual(true);
+    expect(component.valueLabelElement.text()).toEqual('0');
+    expect(component.valueLabelElement.prop('aria-hidden')).toEqual('true');
+    expect(component.maxLabelElement.exists()).toEqual(true);
+    expect(component.maxLabelElement.text()).toEqual('100ETH');
+    expect(component.maxLabelElement.prop('aria-hidden')).toEqual('true');
+    expect(component.sliderElement.exists()).toEqual(true);
+    expect(component.sliderElement.hasClass(css.slider)).toEqual(true);
+    expect(component.sliderElement.prop('min')).toEqual(0);
+    expect(component.sliderElement.prop('max')).toEqual(100);
+    expect(component.sliderElement.prop('value')).toEqual(0);
+    expect(component.sliderElement.prop('step')).toEqual(1);
+    expect(component.sliderElement.prop('aria-atomic')).toEqual(true);
+    expect(component.sliderElement.prop('aria-valuemin')).toEqual(0);
+    expect(component.sliderElement.prop('aria-valuemax')).toEqual(100);
+    expect(component.sliderElement.prop('aria-valuenow')).toEqual(0);
+    expect(component.sliderElement.prop('aria-label')).toEqual('Current allocation: 0 ETH');
+    expect(component.sliderElement.prop('type')).toEqual('range');
+    expect(component.sliderElement.prop('onChange')).toBeInstanceOf(Function);
+    expect(component.sliderElement.prop('onKeyDown')).toBeInstanceOf(Function);
+  });
+
+  describe('can change its value with the keyboard', () => {
+    const keyboardCases = [
+      ['increasing', 'ArrowUp', 1],
+      ['increasing', 'ArrowRight', 1],
+      ['increasing', 'PageUp', 10],
+      ['decreasing', 'ArrowDown', -1],
+      ['decreasing', 'ArrowLeft', -1],
+      ['decreasing', 'PageDown', -10]
+    ];
+
+    beforeEach(() => {
+      component = mockSlider({initialValue: 50});
+    });
+
+    it.each(keyboardCases)('%s it by %i when pressing %s', (_, key, numberChange) => {
```

Comment:
niiice

Reply:
I actually wanted to use the [TST syntax](https://jestjs.io/docs/en/api#2-testeach-table-name-fn-timeout) but `prettier` kept destroying it.